bin/run

```diff
@@ -81,33 +92,47 @@ with gfile.GFile(model_file, 'rb') as f:
         })
 
 inputs = []
-for i, spec in enumerate(input_spec):
-    def scope(i: int, spec: dict) -> Callable[[], tf.Tensor]:
-        def provide_input() -> tf.Tensor:
-            pl = tf.placeholder(tf.float32, shape=spec['shape'], name="api/{}".format(i))
-            return pl
-
-        return provide_input
-
-    inputs.append(scope(i, spec))
-
 
-if config.input_file != '':
-    # TODO: Currently this allows for only one input
-    # If we want to load models with multiple diffrent inputs
-    # We will need to save our inputs as a list of np.array
-    input_data = [np.load(config.input_file)]
-    assert len(input_data) == len(input_spec), 'the input_data length ({}) should be equal the number of inputs required by the loaded model ({})'.format(
-        len(input_data), len(input_spec))
+# If data_prep_path is set then extract data-preprocessing
+# graph from the SavedModel. Finally run it on the input-provider
+if config.data_prep_path:
+    data_file_names = list_files_from_dir(config.data_dir)
+    def provide_input():
+            return data_prep_from_saved_model(config.data_prep_path, 
+                data_file_names,
+                config.batch_size,
+                config.data_prep_start_node,
+                config.data_prep_end_node)
+
+    inputs.append(provide_input)
 else:
-    input_data = []
-    for spec in input_spec:
-        input_data.append(
-            np.random.standard_normal(spec['shape']).tolist()
+    for i, spec in enumerate(input_spec):
+        def scope(i: int, spec: dict) -> Callable[[], tf.Tensor]:
```

Comment:
I would consider defining `scope` outside the loop, just to avoid the name hiding

Reply:
Agreed — this whole loop can be made a list comprehension that way